README.md

```diff
@@ -273,7 +278,103 @@ should mention how to instantiate those variables/files, etc.
+
+That doesn’t sound like much fun, does it?
+
+This is where automated testing comes in. Automated testing is the execution of your test plan (the parts of your application you want to test, 
+the order in which you want to test them, and the expected responses) by a script instead of a human. Software testing involves the execution of a 
+software component or system component to evaluate one or more properties of interes. In general, these properties indicate the extent to which the component
+ or system under test:
+
+* meets the requirements that guided its design and development,
+* responds correctly to all kinds of inputs,
+* performs its functions within an acceptable time,
+* is sufficiently usable,
+* can be installed and run in its intended environments, and
+* achieves the general result its stakeholders desire.
+
+#### Test driven development
+Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle: requirements are turned into 
+very specific test cases, then the code is improved so that the tests pass. In simple terms, 
+test cases for each functionality are created and tested first and if the test fails then the new code is written in order to pass the test 
+and making code simple and bug-free.
+
+
+
+Useful links for writing tests in our most common  languages and frameworks: 
+* [Python](https://realpython.com/python-testing/)
+* [React](https://reactjs.org/docs/testing-recipes.html)
+* [Javascript](https://jestjs.io/)
+* [R](https://r-pkgs.org/tests.html)
+
+## Continuous integration and delivery
+
+The CI/CD pipeline is one of the best practices for devops teams to implement, 
+for delivering code changes more frequently and reliably. It is also an agile methodology 
+best practice, as it enables software development teams to focus on meeting business requirements, 
+and code quality because deployment steps are automated.
+
+### Continuous integration
+Developers practicing continuous integration merge their changes back to the main branch as often as possible. 
+The developer's changes are validated by creating a build and running automated tests against the build. 
+By doing so, you avoid the integration hell that usually happens when people wait for release day to merge
+ their changes into the release branch.
+
+Continuous integration puts a great emphasis on testing automation to check that the application is not broken 
+whenever new commits are integrated into the main branch.
+
+#### What you need (cost)
+Your team will need to write automated tests for each new feature, improvement or bug fix.
+You need a continuous integration server that can monitor the main repository and run the tests automatically for every new commits pushed.
+Developers need to merge their changes as often as possible, at least once a day.
```

Comment:
This seems unrealistic. How about something about encouraging developers to limit the size of changes to make them easier to review and merge and working to merge changes as soon as they're ready?

Reply:
yeah, I agree.